### PR TITLE
Highlight word in reading mode as well 

### PR DESCRIPTION
**Pull request by naveed-ahmad**
Well I've an idea, we've segments for all verses thanks to @cpfair how about playing audio within its timing boundary when user click on it? If that's possible maybe we don't need word by word audio also we can achieve same features in multiple voices.

Although teaching style recitation will be perfect for word by word audio.

- [x]  If no verse is being playing, and user click on word play the word and stop audio again.
- [x]  If an ayah is playing and user click on word within ayah, play the word and continue playing ayah from last playing word
- [x]  If ayah is playing and user click on word from any other ayah, play that word and ...? 

## Reading mode enhancements 
* Highlight current playing word
* Fixed tooltip
* Added auto scroll to playing ayah

## Refactoring 
* DRYed rendering on words, line and ayah used same word compontnet now

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33086

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33087

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33088

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33089

**Comment by ahmedre** [comment]
rebuild

File: src/components/Ayah/index.js [hunk 1]
```diff
@@ -62,16 +62,31 @@ export default class Ayah extends Component {
...

+
+    
if(isCurrentAyah && isPlaying) {

+       audioActions.setCurrentWord(word.dataset.key)
;
+    }

+    else {
```
Comment (sabeurthabti) [inline src/components/Ayah/index.js]: same line

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33095

File: src/components/Ayah/index.js [hunk 2]
```diff
@@ -62,16 +62,31 @@ export default class Ayah extends Component {
...
+       audioActions.setAyah(word.dataset.ayah);
+       audioActions.playCurrentWord(word.dataset.key);
+       //audioActions.play();
```
Comment (sabeurthabti) [inline src/components/Ayah/index.js]: remove?

Comment (naveed-ahmad) [inline src/components/Ayah/index.js]: sure

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33096

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33097

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33102

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33104

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33106

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33107

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33111

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33117

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33120

**Comment by sabeurthabti** [comment]
LGTM?

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33123

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33125

**Pull request merged**